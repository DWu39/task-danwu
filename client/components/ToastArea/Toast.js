import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import Button from '../Button';
import theme from '../../style/theme';

const StyledToast = styled.div`
  background: ${({ success }) => success ? theme.greenLight : theme.redLight};
  border: 1px solid;
  border-color: ${({ success }) => success ? theme.green : theme.red};
  border-radius: 4px;
  color: ${({ success }) => success ? theme.green : theme.red};
  margin: 12px;
  padding: 8px 20px;

  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const StyledToastText = styled.p`
  margin: 0;
`;

const CloseButton = styled(Button)`
  background: none;
  color: ${({ success }) => success ? theme.green : theme.red};
  font-size: 16px;
  padding: 0;
`;

const Toast = ({
  onClickClose,
  success,
  text
}) => (
  <StyledToast success={success}>
    <StyledToastText>{text}</StyledToastText>
    <CloseButton onClick={onClickClose} success={success}>
      <i className="fa fa-times"></i>
    </CloseButton>
  </StyledToast>
);

Toast.propTypes = {
  onClickClose: PropTypes.func,
  success: PropTypes.bool,
  text: PropTypes.string
}

Toast.defaultProps = {
  onClickClose: null,
  success: false,
  text: ''
}

export default Toast;
