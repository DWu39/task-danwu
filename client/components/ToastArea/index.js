import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import Toast from './Toast';
import theme from '../../style/theme';

const StyledToastArea = styled.section`
  bottom: 50px;
  position: fixed;
  right: 50px;
  width: 35%;
`;

const ToastArea = ({
  closeToast,
  toasts
}) => {
  return (
    <StyledToastArea>
      {toasts.length
        ? toasts.map((toast) => (
          <Toast
            key={toast.id}
            onClickClose={() => { closeToast(toast.id); }}
            {...toast}
          />
        ))
        : null
      }
    </StyledToastArea>
  );
};

ToastArea.propTypes = {
  closeToast: PropTypes.func,
  toasts: PropTypes.array
}

ToastArea.defaultProps = {
  closeToast: null,
  toasts: []
}

export default ToastArea;
