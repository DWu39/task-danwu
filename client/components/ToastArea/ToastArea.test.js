import React from 'react';
import { shallow } from 'enzyme';

import ToastArea from './index';

const mockToasts = [
  {
    id: 'id1',
    success: true,
    text: 'win!'
  },
  {
    id: 'id2',
    text: 'fail!'
  }
];

describe('ToastArea', () => {
  it('should render empty correctly', () => {
    const output = shallow(<ToastArea/>);
    expect(output).toMatchSnapshot();
  });

  it('should render toasts correctly', () => {
    const output = shallow(<ToastArea toasts={mockToasts}/>);
    expect(output).toMatchSnapshot();
  });
});
