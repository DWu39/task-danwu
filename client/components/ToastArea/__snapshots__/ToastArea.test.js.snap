// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastArea should render empty correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToastArea
    closeToast={null}
    toasts={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": null,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ToastArea should render toasts correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ToastArea
    closeToast={null}
    toasts={
      Array [
        Object {
          "id": "id1",
          "success": true,
          "text": "win!",
        },
        Object {
          "id": "id2",
          "text": "fail!",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Toast
          id="id1"
          onClickClose={[Function]}
          success={true}
          text="win!"
        />,
        <Toast
          id="id2"
          onClickClose={[Function]}
          success={false}
          text="fail!"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "id1",
        "nodeType": "function",
        "props": Object {
          "id": "id1",
          "onClickClose": [Function],
          "success": true,
          "text": "win!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "id2",
        "nodeType": "function",
        "props": Object {
          "id": "id2",
          "onClickClose": [Function],
          "success": false,
          "text": "fail!",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Toast
            id="id1"
            onClickClose={[Function]}
            success={true}
            text="win!"
          />,
          <Toast
            id="id2"
            onClickClose={[Function]}
            success={false}
            text="fail!"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "id1",
          "nodeType": "function",
          "props": Object {
            "id": "id1",
            "onClickClose": [Function],
            "success": true,
            "text": "win!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "id2",
          "nodeType": "function",
          "props": Object {
            "id": "id2",
            "onClickClose": [Function],
            "success": false,
            "text": "fail!",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
